cli/azd/pkg/devcenter/manager.go

```diff
@@ -265,15 +268,19 @@ func (m *manager) LatestArmDeployment(
 		tagEnvTypeName, envTypeOk := d.Tags[DeploymentTagEnvironmentType]
 		tagEnvName, envOk := d.Tags[DeploymentTagEnvironmentName]
 
-		if !devCenterOk || !projectOk || !envTypeOk || !envOk {
-			return false
-		}
+		// ARM runner deployments contain the deployment tags for the specific environment
+		isArmDeployment := devCenterOk && *tagDevCenterName == m.config.Name &&
+			projectOk && *tagProjectName == m.config.Project &&
+			envTypeOk && *tagEnvTypeName == m.config.EnvironmentType &&
+			envOk && *tagEnvName == env.Name
 
-		if *tagDevCenterName == m.config.Name ||
-			*tagProjectName == m.config.Project ||
-			*tagEnvTypeName == m.config.EnvironmentType ||
-			*tagEnvName == env.Name {
+		// Support for untagged Bicep ADE deployments
+		// If the deployment is not tagged but starts with the current date and is running
+		// this is another indication that this is the latest running Bicep deployment
+		isBicepDeployment := !isArmDeployment &&
+			strings.HasPrefix(*d.Name, fmt.Sprintf("%s-", time.Now().UTC().Format("2006-01-02")))
```

Comment:
This seems very fragile to me.  We have to match the exact date?  Or are we just trying to find the newest thing that looks like it starts with a date in this format and then a dash?

It feels like at best this means `azd` works correctly one day per Bicep ADE deployment?  That doesn't seem ideal.

Reply:
Yes, this is doing a name match looking for a deployment with today's date.  This could fail on some env refresh scenarios that I can look at.  Otherwise we could just look for a date pattern instead of today's date.

Reply:
Here is an example of a running deployment.
<img width="881" alt="image" src="https://github.com/Azure/azure-dev/assets/6540159/972d8d27-0b66-4144-870a-4a3a5d7540c3">

Reply:
@ellismg - updated to use a more generic regex to find date like strings that satisfy the ADE bicep deployments.